web/modules/custom/asset_release/src/Controller/AssetReleaseController.php

```diff
@@ -73,10 +99,7 @@ public function add(RdfInterface $rdf_entity) {
    *   The access result object.
    */
   public function createAssetReleaseAccess(RdfInterface $rdf_entity) {
-    $user = $this->currentUser();
-    $membership = Og::getMembership($user, $rdf_entity);
-    // @todo: Remove check for empty after ISAICP-2369 is in.
-    return (!empty($membership) && $membership->hasPermission('create asset_release rdf_entity')) ? AccessResult::allowed() : AccessResult::forbidden();
+    return $this->ogAccess->userAccessEntity('create', $this->createNewAssetRelease($rdf_entity), $this->currentUser());
```

Comment:
I don't like this. You are creating a new temporary asset release with the group set to the solution to check the permissions, and then discarding it straight away, since a new fresh asset release is going to be created to be used in the form, where again the group is set (plus other values).
At this point I think we should simply use something like this:
`return $this->ogAccess->userAccess($rdf_entity, 'create asset_release rdf_entity')`
which avoids temporary entities.